soroban-env-host/src/host.rs

```diff
@@ -2756,13 +2757,18 @@ impl VmCallerCheckedEnv for Host {
             for signer in signers {
                 if let SignerKey::Ed25519(ref this_signer) = signer.key {
                     if &target_signer == this_signer {
+                        // Clamp the weight at 255. Stellar protocol before v10
+                        // allowed weights to exceed 255, but the max threshold
+                        // is 255, hence there is no point in having a larger
+                        // weight.
+                        let weight = min(signer.weight, u8::MAX as u32);
                         // We've found the target signer in the account signers, so return the weight
-                        return Ok(signer.weight.into());
+                        return Ok(weight.into());
```

Comment:
👍🏻 Oh this is great, this means we won't need to handle > 255 signers in the SDK and elsewhere 👏🏻, great idea.

Reply:
Sounds good, I wish we could do that on host too, but currently we don't have `u8` RawVal...

Reply:
No need, for u8 values we transmit then as u32 and assume they will always be < 255 on the guest side.

Reply:
I realize that we can transfer these as u32, but that also reduces type safety in host, so we have to rely only on documentation. That's just a random thought - I'm not calling for any changes here now.

Reply:
💯. It's a problem across the board, but we resolved at some point to have the SDK always trust the host, and so the SDK makes a ton of assumptions about types. For example, it does conversions from `Object` to types without checking their type code too.